File: lingting-spring-web/src/main/java/live/lingting/spring/web/converter/Converter.java
package live.lingting.spring.web.converter;

import org.springframework.core.convert.TypeDescriptor;
import org.springframework.core.convert.converter.ConditionalGenericConverter;
import org.springframework.util.StringUtils;

/**
 * @author lingting 2022/9/28 11:14
 */
public interface Converter<T> extends ConditionalGenericConverter {

	default String[] toArray(Object source) {
		if (source == null) {
			return new String[0];
		}
		String string = ((String) source).trim();

		if (string.startsWith("[") && string.endsWith("]")) {
			string = string.substring(1, string.length() - 1);
		}
		return StringUtils.commaDelimitedListToStringArray(string);
	}

	@Override
	T convert(Object source, TypeDescriptor sourceType, TypeDescriptor targetType);

}

File: lingting-spring-web/src/main/kotlin/live/lingting/spring/web/converter/Converter.kt
package live.lingting.spring.web.converter

import org.springframework.core.convert.TypeDescriptor
import org.springframework.core.convert.converter.ConditionalGenericConverter
import org.springframework.util.StringUtils

/**
 * @author lingting 2022/9/28 11:14
 */
interface Converter<T> : ConditionalGenericConverter {
    fun toArray(source: Any): Array<String> {
        if (source == null) {
            return arrayOfNulls<String>(0)
        }
        var string = (source as String).trim { it <= ' ' }

        if (string.startsWith("[") && string.endsWith("]")) {
            string = string.substring(1, string.length - 1)
        }
        return StringUtils.commaDelimitedListToStringArray(string)
    }

    override fun convert(source: Any, sourceType: TypeDescriptor, targetType: TypeDescriptor): T
}
